providers/zohocrm/metadata.go

```diff
@@ -0,0 +1,82 @@
+package zohocrm
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/naming"
+)
+
+// restMetadataEndpoint is the resource for retrieving metadata details.
+const restMetadataEndpoint = "settings/fields"
+
+// apiKeyField is the key holding the metadata field name.
+const apiKeyField = "api_name"
+
+type metadataFields struct {
+	Fields []map[string]any `json:"fields"`
+}
+
+func (c *Connector) ListObjectMetadata(ctx context.Context,
+	objectNames []string,
+) (*common.ListObjectMetadataResult, error) {
+	if len(objectNames) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	objMetadata := common.ListObjectMetadataResult{
+		Result: make(map[string]common.ObjectMetadata),
+		Errors: make(map[string]error),
+	}
+
+	for _, obj := range objectNames {
+		url, err := c.getAPIURL(restMetadataEndpoint)
```

Comment:
Since these calls in loop doesn't need to be in order, we can do this is go func and use wait group to optimize time.